### DISCUSSION TITLE
Rename `obslog` (obsolescence log)

### DISCUSSION THREAD

**Discussion by Cretezy** [jj-vcs/jj#3592] [Ideas]
The [obslog](https://martinvonz.github.io/jj/v0.16.0/cli-reference/#jj-obslog) subcommand has a confusing name. It's derived from Mercurial's obsolescence markers, but does not reflect what it does (show other revisions of the same change-id) clearly.

More discussion: https://discord.com/channels/968932220549103686/969291218347524238/1201539793046208532

Potential ideas: 
- `jj changelog` or if split into other subcommands, `jj change`. Overlap with the concept of a changelog
- `jj evolution` or `jj evolutionlog`
- `jj rewritelog`
- `jj editlog`

**Comment by joyously** [comment]
Maybe change the help text to use that phrase "show other revisions of the same change-id", since I couldn't determine what the command did from the help either.
Please don't use `change`, since it is both a verb and noun.
Maybe `overview` or `rev-view` or just `view` since without a `-r` you get `@` which isn't that great usually.
Even though I think commands should be verbs, I think `evolution` is pretty good. 
Does this concept affect all backends?

**Reply by Cretezy** [reply]
I've updated https://github.com/martinvonz/jj/pull/3593 to include a change to the help text

**Comment by noahmayr** [comment]
I kind of like `changelog` but there have been previous discussions whether change / change id should be renamed to something else because that naming has also been confusing.
There's also `revlog` (could be confused for `reflog` though)

**Reply by chriskrycho** [reply]
Interesting thought on `revlog`—that “adjacency” to `reflog` might actually be a positive? Also, Jujutsu could totally have a `did you mean?` if you type `jj reflog`.

**Reply by martinvonz** [reply]
"Revlog" is also what Mercurial's main storage format is called (https://wiki.mercurial-scm.org/Revlog). Just FYI; I don't think that's much reason to avoid it.

**Comment by ilyagr** [comment]
I like `jj rewritelog`. `jj evolutionlog` could also work. It could have `evolog` as an alias.

For `jj changelog`, a user might expect that to convert repo history into a textual changelog for a release.

**Reply by Cretezy** [reply]
I too like `jj rewritelog`.

Maybe `rewlog` could be an alias too? `jj reflog`/`jj revlog` could show "Did you mean rewlog" as @chriskrycho [mentioned](https://github.com/martinvonz/jj/discussions/3592#discussioncomment-9263615).

**Reply by ilyagr** [reply]
I think `rewlog` looks too much like a typo for my taste. `rewrlog` is a bit better, but then I'd just go with `rewritelog` and rely on the fact that `rewr<tab>` should shell-expand to `rewritelog`.

**Reply by ilyagr** [reply]
(This discussion continues below, see https://github.com/martinvonz/jj/discussions/3592#discussioncomment-9317020 and @thoughtpolice 's comment that replies to)

**Comment by thoughtpolice** [comment]
My thinking is:

- I'd like us to avoid compound abbreviations (`rewlog` or `obslog`), because they suffer from the exact problem this discussion wants to solve. I would literally have no clue what `rewlog` meant.
- I'd like us to avoid compoud *words* because &mdash; and I know this is trite &mdash; they look really bad and ugly! `rewritelog` is just ugly for instance. (They do this in Mercurial but I think there's a related issue at play which is that that `hg` doesn't have subcommands IIRC, so they have to have `hg thingcmd` rather than `hg thing cmd`.)
- I think a single, provocative and unique name is good

So from that point of view, I actually quite like the term `evolution`.

**Reply by joyously** [reply]
Would `ancestry` work? Or should it be a subcommand of `show` or `log` or `view` so that there is a verb?

**Reply by noahmayr** [reply]
we could also use `(show|view|log) predecessors` which is what it's called internally (yes, long word but nothing tab completion can't solve)

**Reply by ilyagr** [reply]
I think `evolution` is a good idea, it's better than `evolutionlog`. `jj change evolution` would be clearer, but we already do `jj split` instead of `jj change split` and so on.

I prefer `evolution` to `ancestry` or `predecessors`, because these make me think of a change's parents, and that usually refers to the commit graph.

I don't actually find `rewritelog` ugly, but maybe I'm held hostage by compound words. Generally, I think compound words should be avoided when possible, but not when it makes things less clear. (In this case, it's arguable whether `rewritelog` is clearer for most people. It makes sense to me, and I still like it, but I'm happy with `evolution`).

**Reply by chriskrycho** [reply]
`jj evolution -r <id>` seems excellent to me, for whatever that’s worth. I agree with everyone who dislikes compound-word commands, and “evolution” was (as I noted on the Discord discussion that spawned this) the first word that came to mind for me for describing what is going on with a change. If you wanted to get a little bit cute (…but in a good way, IMO), you could do `jj evolution --of <id>`, with `--of` as an alias for `-r`/`--revision`. 😅

**Reply by yuja** [reply]
"evolution" sounds like a command namespace (i.e. `jj evolution <verb>`), but I don't have a strong feeling.

**Reply by ilyagr** [reply]
If we ever have other commands to put in that namespace, we could rename `jj evolution` to `jj evolution log` or `jj evolve log`.

**Comment by cenviity** [comment]
`jj obslog` has been renamed to `jj evolog` / `jj evolution-log` in 8fca7a5.
